src/pages/cli/project/troubleshooting.mdx

```diff
@@ -0,0 +1,191 @@
+    +	The GraphQL resolvers for Query, Mutation or Subscription are converted into [VTL (Velocity Template Language)](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-mapping-template-reference-programming-guide.html) files. AWS AppSync uses VTL to translate GraphQL requests from clients into a request to your data source.
+
+  * GraphQL schema client-side code generation: The Amplify CLI GraphQL workflow allows developers to generate client side code for web and mobile clients using the `amplify codegen` command. 
+
+## Troubleshooting `amplify push` failures
+<Callout information>
+Error Messages: “An error occurred during the push operation”
+</Callout>
+
+The [`amplify push`](https://docs.amplify.aws/cli/start/workflows/#amplify-push) command performs the following steps:
+
+1. It generates CloudFormation for deployment of resources to AWS.
+1. When building a GraphQL API, the CLI runs `amplify api gql-compile` internally to compile the schema and generate VTL (Velocity Templates) for mapping resolvers and CloudFormation templates to allocate AWS resources. 
+1. It builds, packages, and uploads the deployment artifacts into the application's deployment bucket. 
+1. It uses the AWS CloudFormation SDK to deploy the CloudFormation stack into your AWS account.
+1. On a successful deployment of the resources to AWS, the Amplify CLI renames the deployment package in the deployment bucket to `#current-cloud-backend.zip`. This file is the source of truth for your deployment.
+1. If your local project is deleted or corrupted, you can always go back to your last successful deployment using [`amplify pull`](https://docs.amplify.aws/cli/start/workflows/#amplify-pull).
+
+![Amplify folders created in the deployment bucket](/images/ts-guide-amplify-deployment-folders.png)
+
+The following section shows some examples on how to debug `amplify push` failures: 
+
+### Push failures from manual changes to cloud resources - Drift 
+<Callout warning>
+**Warning:** Once an application is deployed to the cloud using Amplify CLI the cloud resources like DynamoDB tables, Cognito Roles, IAM policies must never be updated manually. 
+</Callout>
+
+The Amplify CLI cannot track manual changes applied to deployed resources. This state is called “CloudFormation drift” and may cause subsequent deployments to fail. Currently there is no automated way to fix CloudFormation drift. Each resource must be inspected through the AWS console for drift in its CloudFormation stack and the drift has to be resolved manually.
+Assume, you have created a GraphQL schema in Amplify with multiple models which results in the creation of AppSync API models, resolvers and multiple DynamoDB tables and GSIs. 
```

Comment:
Is this sentence in the right place? I don't see the immediate connection to the next sections.